rules/bitstreams.bzl

```diff
@@ -73,6 +73,12 @@ bitstreams_repo = repository_rule(
             default = Label("//rules/scripts:bitstreams_workspace.py"),
             allow_files = True,
         ),
+        # Depend on the current SCM revision so targets in @bitstreams will
+        # never point to artifacts from the wrong commit. This came up in
+        # <https://github.com/lowRISC/opentitan/issues/16832>.
+        "_scm_revision": attr.label(
+            default = Label("//:.scm_revision.txt")
```

Comment:
I don't think we should be depending on a dirtied tree to make builds happen.

Reply:
Would adding it to an already generated file like "stable-status.txt" so the tree isn't dirtied address your concern?

Reply:
I'm mildly worried that volatile-status.txt and stable-status.txt might be special kinds of nodes in the dependency graph.

Reply:
To add some color to that, we already know volatile-status.txt is special. See <https://bazel.build/docs/user-manual#workspace-status>:

> In order to avoid rerunning stamped actions all the time though, Bazel pretends that the volatile file never changes.

Reply:
stable-status.txt would reflect no new dirtying of the tree and nothing unanticipated by bazel's development. It's a little strange to pretend to stamp an artifact, but it is updating the commit sha it uses for HEAD, it could even be made to actually use what it gets from the stamp (though that wouldn't have a real effect).